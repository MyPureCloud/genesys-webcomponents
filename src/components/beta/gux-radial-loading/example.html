<div>
  <h1>Default (Modal)</h1>
  <gux-radial-loading-beta></gux-radial-loading-beta>
</div>

<div>
  <h1>Full Page</h1>
  <gux-radial-loading-beta context="full-page"></gux-radial-loading-beta>
</div>

<div>
  <h1>Input</h1>
  <gux-radial-loading-beta context="input"></gux-radial-loading-beta>
</div>

<div>
  <h1>Modal</h1>
  <gux-radial-loading-beta context="modal"></gux-radial-loading-beta>
</div>
