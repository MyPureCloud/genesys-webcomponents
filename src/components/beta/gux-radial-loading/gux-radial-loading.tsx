import { Component, h, JSX, Prop } from '@stencil/core';

export type GuxRadialLoadingContext = 'full-page' | 'input' | 'modal';

@Component({
  styleUrl: 'gux-radial-loading.less',
  tag: 'gux-radial-loading-beta'
})
export class GuxRadialLoading {
  /**
   * The display context the component is in.
   */
  @Prop()
  context: GuxRadialLoadingContext = 'modal';

  render(): JSX.Element {
    return (
      <div class={`spinner-container ${this.context}`} aria-busy="true">
        <div class="spin-circle" />
      </div>
    );
  }
}
