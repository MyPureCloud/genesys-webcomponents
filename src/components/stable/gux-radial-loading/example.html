<div>
  <h1>Default (Modal)</h1>
  <gux-radial-loading></gux-radial-loading>
</div>

<div>
  <h1>Full Page</h1>
  <gux-radial-loading context="full-page"></gux-radial-loading>
</div>

<div>
  <h1>Input</h1>
  <gux-radial-loading context="input"></gux-radial-loading>
</div>

<div>
  <h1>Modal</h1>
  <gux-radial-loading context="modal"></gux-radial-loading>
</div>
