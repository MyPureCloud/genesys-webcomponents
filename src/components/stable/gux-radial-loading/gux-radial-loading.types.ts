export type GuxRadialLoadingContext = 'full-page' | 'input' | 'modal';
