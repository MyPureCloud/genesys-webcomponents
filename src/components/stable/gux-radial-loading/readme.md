# gux-radial-loading

When the loading of a task is known, this component displays a radial indicator and the completed percentage of
the task.

Otherwise, if the loading is unknown, and no percentage is provided, it instead displays a simple Loading Spinner.

<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                              | Type                                | Default   |
| --------- | --------- | ---------------------------------------- | ----------------------------------- | --------- |
| `context` | `context` | The display context the component is in. | `"full-page" \| "input" \| "modal"` | `'modal'` |


## Dependencies

### Used by

 - [gux-page-loading-spinner](../gux-page-loading-spinner)
 - [gux-radial-progress](../gux-radial-progress)

### Graph
```mermaid
graph TD;
  gux-page-loading-spinner --> gux-radial-loading
  gux-radial-progress --> gux-radial-loading
  style gux-radial-loading fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
