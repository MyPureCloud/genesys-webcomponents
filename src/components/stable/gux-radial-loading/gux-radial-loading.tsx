import { Component, Element, h, JSX, Prop } from '@stencil/core';

import { trackComponent } from '../../../usage-tracking';

import { GuxRadialLoadingContext } from './gux-radial-loading.types';

@Component({
  styleUrl: 'gux-radial-loading.less',
  tag: 'gux-radial-loading'
})
export class GuxRadialLoading {
  @Element()
  private root: HTMLElement;

  /**
   * The display context the component is in.
   */
  @Prop()
  context: GuxRadialLoadingContext = 'modal';

  componentWillLoad(): void {
    trackComponent(this.root, { variant: this.context });
  }

  render(): JSX.Element {
    return (
      <div class={`gux-spinner-container gux-${this.context}`} aria-busy="true">
        <div class="gux-spin-circle" />
      </div>
    );
  }
}
