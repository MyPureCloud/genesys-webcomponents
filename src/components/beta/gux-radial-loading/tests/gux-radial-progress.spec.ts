import { newSpecPage } from '@stencil/core/testing';
import { GuxRadialLoading } from '../gux-radial-loading';

describe('gux-radial-loading', () => {
  let component: GuxRadialLoading;

  beforeEach(async () => {
    const page = await newSpecPage({
      components: [GuxRadialLoading],
      html: `<gux-radial-loading-beta></gux-radial-loading-beta>`,
      language: 'en'
    });

    component = page.rootInstance;
  });

  it('should build', async () => {
    expect(component).toBeInstanceOf(GuxRadialLoading);
  });

  describe('#render', () => {
    [
      '<gux-radial-loading-beta></gux-radial-loading-beta>',
      '<gux-radial-loading-beta context="modal"></gux-radial-loading-beta>',
      '<gux-radial-loading-beta context="full-page"></gux-radial-loading-beta>',
      '<gux-radial-loading-beta context="input"></gux-radial-loading-beta>'
    ].forEach((input, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({
          components: [GuxRadialLoading],
          html: input,
          language: 'en'
        });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
