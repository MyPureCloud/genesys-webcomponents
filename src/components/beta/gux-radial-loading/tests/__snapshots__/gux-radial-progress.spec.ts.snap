// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-radial-loading #render should render component as expected (1) 1`] = `
<gux-radial-loading-beta>
  <div aria-busy="true" class="modal spinner-container">
    <div class="spin-circle"></div>
  </div>
</gux-radial-loading-beta>
`;

exports[`gux-radial-loading #render should render component as expected (2) 1`] = `
<gux-radial-loading-beta context="modal">
  <div aria-busy="true" class="modal spinner-container">
    <div class="spin-circle"></div>
  </div>
</gux-radial-loading-beta>
`;

exports[`gux-radial-loading #render should render component as expected (3) 1`] = `
<gux-radial-loading-beta context="full-page">
  <div aria-busy="true" class="full-page spinner-container">
    <div class="spin-circle"></div>
  </div>
</gux-radial-loading-beta>
`;

exports[`gux-radial-loading #render should render component as expected (4) 1`] = `
<gux-radial-loading-beta context="input">
  <div aria-busy="true" class="input spinner-container">
    <div class="spin-circle"></div>
  </div>
</gux-radial-loading-beta>
`;
